symforce/codegen/backends/python/templates/util/util.jinja

```diff
@@ -217,3 +217,27 @@ def {{ function_name_and_args(spec) }}:
         {%- if not loop.last %}, {% endif %}
     {%- endfor -%}
 {% endmacro %}
+
+{# ------------------------------------------------------------------------- #}
+
+{# Macro to flatten an array if it's an ndarray and is a vector. Also, raises
+ # a ValueError if the length is not equal to size and the shape is not that of
+ # a vector.
+ #
+ # Args:
+ #     array (str): The name of an sequence that might be an ndarray we wish to flatten
+ #     size (int): The expected size of the array
+ #}
+{% macro flatten_if_ndarray(array, size) %}
+if isinstance({{ array }}, numpy.ndarray):
+    if {{ array }}.shape in [({{ size }}, 1), (1, {{ size }})]:
+        {{ array }} = {{ array }}.flatten()
+    elif {{ array }}.shape != ({{ size }},):
+        raise ValueError("Expected {{ array }} to be a vector of length {{ size }}; instead had shape {}".format({{ array }}.shape))
```

Comment:
Should we instead be raising an `IndexError`? In the auto-generated code, it makes sense to raise an `IndexError`, as, since we have no special checks for non-vector matrices, if you pass in, say a matrix which is too small, an `IndexError` would be raised.

It would be strange for a too small vector to result in a different error, so we manually raise an `IndexError there as well.

Similarly, if you set `reshape_vectors=False` in your `PythonConfig`, then for the same reason, an `IndexError` would be automatically raised. It seems preferable for the same error to be raised for the same incorrect input regardless of the value of `reshape_vectors`.

But here is different, as the macro is only used in places where just vectors are expected (so no matrices with different errors), and which are the same regardless of your `PythonConfig`.

Since the type is correct, but the value is invalid, `ValueError` is what I would think would be the correct error to raise. But it seems strange to me that we'd raise `ValueError` in the constructor of a `Sym` class for an incorrectly sized argument, but an `IndexError` in the method of the same class for the same problem.

Reply:
Yep I think we should probably raise IndexError here for consistency

Reply:
I've switched it over to IndexError.